src/replica/test/replica_disk_test.cpp

```diff
@@ -1,122 +1,69 @@
-        stub->on_disk_stat();
-    }
-
-    std::vector<std::shared_ptr<dir_node>> get_fs_manager_nodes()
-    {
-        return stub->_fs_manager._dir_nodes;
-    }
+    void SetUp() override { generate_fake_rpc(); }
 
 private:
-    void generate_mock_app_info()
+    void generate_fake_rpc()
     {
-        app_info_1.app_id = 1;
-        app_info_1.app_name = "disk_test_1";
-        app_info_1.app_type = "replica";
-        app_info_1.is_stateful = true;
-        app_info_1.max_replica_count = 3;
-        app_info_1.partition_count = 8;
-
-        app_info_2.app_id = 2;
-        app_info_2.app_name = "disk_test_2";
-        app_info_2.app_type = "replica";
-        app_info_2.is_stateful = true;
-        app_info_2.max_replica_count = 3;
-        app_info_2.partition_count = 16;
-    }
-
-    void generate_mock_dir_nodes(int num)
-    {
-        int64_t disk_capacity_mb = num * 100;
-        while (num > 0) {
-            int64_t disk_available_mb = num * 50;
-            int disk_available_ratio =
-                static_cast<int>(std::round((double)100 * disk_available_mb / disk_capacity_mb));
-            // create one mock dir_node and make sure disk_capacity_mb_ > disk_available_mb_
-            dir_node *node_disk = new dir_node("tag_" + std::to_string(num),
-                                               "full_dir_" + std::to_string(num),
-                                               disk_capacity_mb,
-                                               disk_available_mb,
-                                               disk_available_ratio);
-
-            int app_id_1_disk_holding_replica_count =
-                app_id_1_primary_count_for_disk + app_id_1_secondary_count_for_disk;
-            while (app_id_1_disk_holding_replica_count-- > 0) {
-                node_disk->holding_replicas[app_info_1.app_id].emplace(
-                    gpid(app_info_1.app_id, app_id_1_disk_holding_replica_count));
-            }
-
-            int app_id_2_disk_holding_replica_count =
-                app_id_2_primary_count_for_disk + app_id_2_secondary_count_for_disk;
-            while (app_id_2_disk_holding_replica_count-- > 0) {
-                node_disk->holding_replicas[app_info_2.app_id].emplace(
-                    gpid(app_info_2.app_id, app_id_2_disk_holding_replica_count));
-            }
-
-            stub->_fs_manager._dir_nodes.emplace_back(node_disk);
-            num--;
-        }
+        // create RPC_QUERY_DISK_INFO fake request
+        dsn::message_ptr fake_query_request = dsn::message_ex::create_request(RPC_QUERY_DISK_INFO);
+        query_disk_info_request query_request;
+        ::dsn::marshall(fake_query_request, query_request);
+        dsn::message_ex *recvd_query_request = fake_query_request->copy(true, true);
+        fake_query_disk_rpc =
+            rpc_holder<query_disk_info_request, query_disk_info_response>::auto_reply(
+                recvd_query_request);
```

Comment:
This is not the standard way to construct a fake rpc.

```cpp
using query_disk_info_rpc = rpc_holder<query_disk_info_request, query_disk_info_response>
auto query_request = dsn::make_unique<query_disk_info_request>();
query_disk_info_rpc  rpc(std::move(query_request), RPC_QUERY_DISK_INFO);
```

You should not enable "auto_reply" of the RPC. Because in a unit-test, you should never expect when the RPC finishes, the test will respond to any real client.